From afa739f1e63fa3959f974fa3c196e9449d66864e Mon Sep 17 00:00:00 2001
From: NIIBE Yutaka <gniibe@fsij.org>
Date: Wed, 30 Apr 2014 17:07:44 +0900
Subject: [PATCH 20/52] * src/pam/auth-support/getpin-cb.c (query_user): Remove
 checking of digits. Change error message when it's too short.

---
 src/pam/auth-support/getpin-cb.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/pam/auth-support/getpin-cb.c b/src/pam/auth-support/getpin-cb.c
index 10acbe7..5dfba9a 100644
--- a/src/pam/auth-support/getpin-cb.c
+++ b/src/pam/auth-support/getpin-cb.c
@@ -81,9 +81,10 @@ query_user (poldi_ctx_t ctx, const char *info, char *pin, size_t pin_size)
 	 Shouldn't they be done in scdaemon itself?  -mo */
 
       if (strlen (buffer) < 6)	/* FIXME? is it really minimum of 6 bytes? */
-	log_msg_error (ctx->loghandle, _("invalid PIN"));
-      else if (!all_digitsp (buffer))
-	log_msg_error (ctx->loghandle, _("invalid characters in PIN"));
+	{
+	  log_msg_error (ctx->loghandle, _("PIN too short"));
+	  conv_tell (ctx->conv, "%s", _("PIN too short"));
+	}
       else
 	break;
     }
-- 
2.17.0

